akka-typed/src/main/scala/akka/typed/cluster/ClusterSingletonManager.scala

```diff
@@ -0,0 +1,103 @@
+  val singletonIdentificationInterval: FiniteDuration,
+  val removalMargin:                   FiniteDuration,
+  val handOverRetryInterval:           FiniteDuration,
+  val bufferSize:                      Int) extends NoSerializationVerificationNeeded {
+
+  private[akka] def toManagerSettings: ClusterSingletonManagerSettings = ???
+  private[akka] def toProxySettings: ClusterSingletonProxySettings = ???
+
+}
+
+object ClusterSingletonManager extends ExtensionId[ClusterSingletonManager] {
+
+  override def createExtension(system: ActorSystem[_]): ClusterSingletonManager =
+    new ClusterSingletonManagerImpl(system)
+}
+
+private[akka] class ClusterSingletonManagerImpl(system: ActorSystem[_]) extends ClusterSingletonManager {
+  require(system.isInstanceOf[ActorSystemAdapter[_]], "only adapted actor systems can be used for the typed cluster singleton")
+
+  private val untypedSystem = ActorSystemAdapter.toUntyped(system)
+
+  override def spawn[A](
+    behavior:           Behavior[A],
+    name:               String,
+    props:              Props,
+    settings:           ClusterSingletonSettings,
+    terminationMessage: Option[A]                = None) = {
+    // just playing around a bit with how it could look here
+    val adaptedProps = PropsAdapter(behavior, props)
+    val managerName = s"singleton-manager-${name}"
+
+    val cluster = Cluster(system)
+    def shouldRunManager =
+      (settings.role.isEmpty || cluster.self.roles(settings.role.get)) &&
+        (settings.dataCenter.isEmpty || cluster.self.dataCenter == settings.dataCenter.get)
+
+    if (shouldRunManager) {
+      // start singleton on this node
+      untypedSystem.actorOf(
+        OldSingletonManager.props(adaptedProps, terminationMessage, settings.toManagerSettings),
+        managerName)
+    }
+
+    // start proxy
+    // Alternative idea, put a common supervisor for manager and proxy rather than using two places in the tree,
+    // not sure what the gain would be though.
+    val untypedProxy = untypedSystem.actorOf(
+      ClusterSingletonProxy.props(s"/user/$managerName", settings.toProxySettings),
+      s"singleton-proxy-${name}")
+
+    ActorRefAdapter[A](untypedProxy)
+  }
+}
+
+trait ClusterSingletonManager extends Extension {
+
+  // Ideas:
+  //  1. single method creates both manager and proxy
+  //  2. method signature mimics the general actor starting signature and name
+  // if needed we could also provide a spawnManager and a spawnProxy (but I think those are only ever used
+  // to do the same kind of if-else-with-role/dc in user logic)
+  /**
+   * Start if needed and provide a proxy to a named singleton
+   *
+   * @return A proxy actor that can be used to communicate with the singleton in the cluster
+   */
+  def spawn[A](
+    behavior:           Behavior[A],
+    name:               String,
+    props:              Props,
+    settings:           ClusterSingletonSettings,
+    terminationMessage: Option[A]                = None
```

Comment:
I have slightly mixed feelings about this, since some of the settings/parameters are specific to the proxy and some to the manager/singleton.

On the other hand I like the simplification.

a slightly confusing thing, what if I for some reason would like to define the Props (e.g. dispatcher) for the proxy?

Reply:
Since the `ClusterSingletonSettings` essentially is a higher level factory for the two old settings types, we could probably provide those respective APIs as well for special cases. 

(However when I was thinking about it I could not think of use cases that seemed useful enough that it felt worth prototyping such methods.)